File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/oauth_authentication.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import OAuthCallbackForm, OAuthLoginRequestForm, OAuthLoginResponse, TokenDTO

class OauthAuthenticationApi:
    """oauth_authentication API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def get_auth_url(self, body: OAuthLoginRequestForm) -> OAuthLoginResponse:
        """Get OAuth authorization URL"""
        return self._client.post(f"/backend/v3/api/auth/oauth/get_auth_url", json=body)

    def create_get_auth_url(self, body: OAuthLoginRequestForm) -> OAuthLoginResponse:
        """Get OAuth authorization URL"""
        return self._client.post(f"/backend/v3/api/auth/oauth/authorize", json=body)

    def handle_callback(self, body: OAuthCallbackForm) -> TokenDTO:
        """Handle OAuth callback"""
        return self._client.post(f"/backend/v3/api/auth/oauth/callback", json=body)

File: sdkwork-backend-sdk-flutter/lib/src/api/oauth_authentication.dart
import '../http/client.dart';
import '../models.dart';

class OauthAuthenticationApi {
  final HttpClient _client;
  
  OauthAuthenticationApi(this._client);

  /// Get OAuth authorization URL
  Future<OAuthLoginResponse?> getAuthUrl(OAuthLoginRequestForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/auth/oauth/get_auth_url'), body: body, contentType: 'application/json');
    return response is OAuthLoginResponse ? response : null;
  }

  /// Get OAuth authorization URL
  Future<OAuthLoginResponse?> createGetAuthUrl(OAuthLoginRequestForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/auth/oauth/authorize'), body: body, contentType: 'application/json');
    return response is OAuthLoginResponse ? response : null;
  }

  /// Handle OAuth callback
  Future<TokenDTO?> handleCallback(OAuthCallbackForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/auth/oauth/callback'), body: body, contentType: 'application/json');
    return response is TokenDTO ? response : null;
  }
}
